### DISCUSSION TITLE
Calibration Help

### DISCUSSION THREAD

**Discussion by 51av0sh** [ESPresense/ESPresense-companion#422] [General]
I've been trying to get better accuracy for the past week tweaking different factors and just wanted to see if anyone has any recommendations to try. I've already remapped the entire house so I'm pretty confident in the floor plan and node locations. I'm very confident in the x,y coordinates for the bounds but the z is throwing me off. We provide 2 points and I wanted to double check, for floor 1 I'm using z=0 on point 1 and z=celling height for point 2. Is this right? For floor 2 I'm starting my z at the 1st floor celling height plus the floor thickness.

Accuracy is mostly great (except in some rooms) but short lived since there is too much jumping between floors and rooms. It's basically pretty accurate for a few minutes and then I assume calibration changes something that throws it all off and then I'm jumping either between floors or between rooms (sometimes both). I have also tweaked the RSSI@1m for 3 main devices I'm tracking. 

My errors look to be too high as well. Any recommendations what I should look at next?

![Screenshot 2023-11-22 203413](https://github.com/ESPresense/ESPresense-companion/assets/85576728/54492aab-72ae-4153-8659-901370083bd4)

**Comment by 51av0sh** [comment]
Is there any documentation on how to read the calibration charts?

**Comment by DTTerastar** [comment]
What you're seeing is pretty much the "state of the art" as far as companion goes.  It's tantalizingly close to working.  The floor identification is the biggest issue.  If you think it's optimization you can turn it off and see if that makes things stabilize. #310 has a paper that i've been reading to try and improve floor identification.  The other idea is to use a x,y,z kalman filter to try and fix the jumping.

**Reply by 51av0sh** [reply]
Yes good call, I'll try that to confirm. I assumed it's the optimization function since I don't have the problem on vanilla espresense (have been rock solid for two years with just minimal jumping between rooms in 2 different houses). But man you're right, companion is so damn close.

**Reply by DTTerastar** [reply]
We are actively working on making this better.  We're adding a variance calculation to only trust low variance nodes.  Stay tuned.

**Comment by jl-p** [comment]
i dont see any device here
